packages/bee-q/src/components/notification/bq-notification.tsx

```diff
@@ -0,0 +1,169 @@
+  // ====================
+
+  private buttonElem: HTMLElement;
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqNotificationElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  @State() private hasBtn = false;
+  @State() private isHidden = true;
+
+  // Public Property API
+  // ========================
+
+  /** Type of Notification */
+  @Prop({ reflect: true }) type: string;
+
+  /** Set property if you want Notification icon to be shown. */
+  @Prop({ reflect: true }) showIcon: boolean;
+
+  /** Set the subject color if you don't want to be black. Subject color will also apply to Icon color if there is one. */
+  @Prop({ reflect: true }) subjectColor: string;
+
+  /** Set property if to false if you want to hide Close icon */
+  @Prop({ reflect: true }) showClose: boolean;
+
+  // Prop lifecycle events
+  // =======================
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  // Listeners
+  // ==============
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  /**
+   * Trigger function when you want to close Notification
+   */
+  @Method()
+  async hideNotification() {
+    this.closeNotification();
+  }
+
+  /**
+   * Trigger function when you want to show Notification
+   */
+  @Method()
+  async showNotification() {
+    this.setNotificationOpened();
+  }
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private getNotificationIconName = () => {
+    if (!this.showIcon) return '';
+
+    const type = this.type;
+    let icon = 'info';
+
+    if (type === 'success') {
+      icon = 'check-circle';
+    }
+
+    if (type === 'error') {
+      icon = 'x-circle';
+    }
+
+    return icon;
+  };
+
+  private handleSlotChange = () => {
+    this.hasBtn = hasSlotContent(this.buttonElem);
+  };
+
+  private closeNotification = () => {
+    this.isHidden = true;
+  };
+
+  private setNotificationOpened = () => {
+    this.isHidden = false;
+  };
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    const styles = {
+      ...(this.subjectColor && { color: getColorCSSVariable(this.subjectColor) }),
+    };
+
+    return (
+      <Host aria-hidden={this.isHidden} hidden={this.isHidden}>
+        <div
+          class="bg-white notification-shadow notification-radius inline-block w-auto px-5 py-4"
```

Comment:
Padding for the component content should 16px all over according to Design specs:
![CleanShot 2023-03-23 at 12 14 12](https://user-images.githubusercontent.com/328492/227172381-acf5c497-b09d-4ad3-8204-0633cc42f787.png)

I would recommend setting a component custom property for that:
```
--bq-notification--padding: theme('padding.4');
```

and then you can use it in JSX as:
```
class="p-[var(--bq-notification--padding)]"
```